### PR TITLE
chore(docs): Updated the Navigation demos to use the masthead instead of the page header

### LINKED ISSUE #9015: Navigation demos - update usage of PageHeader
Possible followup to #8868

Some Navigation (possibly other) demos are still using the deprecated PageHeader component. May want to look into updating these demos to use the Masthead/follow other full page demos.

### PR DESCRIPTION
**Pull request by tlabaj**
**What**: Closes #9015

### CODE REVIEW

**Review by thatblindgeye — COMMENTED** [review]
For the Horizontal Nav demo, the breakpoints aren't kicking in at the right time resulting in content overflowing to the right when the viewport is shrunk down:

Horizontal demo overflow

The Horizontal Nav with Horizontal Subnav demo also has a similar overflow issue, except it's overflowing on page load. 

We could adjust the `visibility` of the ToolbarGroup's and set a width for the Nav components in these two examples.

**Review by mcarrano — CHANGES_REQUESTED** [review]
This still has the issue of the user menu having a lighter background in dark theme.

Screenshot 2023-06-02 at 3 45 29 PM

@mcoker I thought that was fixed?

**Comment by mcoker** [comment]
@mcarrano I have a PR up that fixes it, but it didn't go in until late last night and it was too late to merge - https://github.com/patternfly/patternfly/pull/5643. If we want to try and get it in, I have an open question in the issue here - https://github.com/patternfly/patternfly/pull/5643#issuecomment-1572960124

**Comment by mcarrano** [comment]
Thanks @mcoker I would try to get this in if we can. As to your question on https://github.com/patternfly/patternfly/pull/5643, I'm not sure what the right answer is and want to get input from @mceledonia . Will reach out on Slack to resolve this ASAP.

@nicolethoen can we still try to get this into the release? The user menu does not look correct without it in dark mode.

**Comment by tlabaj** [comment]
> 

@thatblindgeye @mcoker it looks like we are missing a "pf-m-overflow-container" on the toolbar item that holds the nav.  React does not currently support that.  I can add it as part of this PR.

**Comment by tlabaj** [comment]
> > 
> 
> @thatblindgeye @mcoker it looks like we are missing a "pf-m-overflow-container" on the toolbar item that holds the nav. React does not currently support that. I can add it as part of this PR.  It should fix that issue @thatblindgeye  mentions above

File: packages/react-core/src/components/Toolbar/ToolbarGroup.tsx [hunk 1]
```diff
@@ -53,6 +53,8 @@ export interface ToolbarGroupProps extends Omit<React.HTMLProps<HTMLDivElement>,
   };
   /** Content to be rendered inside the data toolbar group */
   children?: React.ReactNode;
+  /** Flag that modifies the toolbar item to hide overflow and respond to available space. Used for horizontal navigation. */
```
Comment (thatblindgeye) [inline packages/react-core/src/components/Toolbar/ToolbarGroup.tsx:56]:
```suggestion
  /** Flag that modifies the toolbar group to hide overflow and respond to available space. Used for horizontal navigation. */
```

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
Just a quick update below

**Review by thatblindgeye — APPROVED** [review]

**Comment by mcarrano** [comment]
User menu looks good now. Just seeing a couple of additional issues.

* The drilldown nav example seems to be broken: https://patternfly-react-pr-9223.surge.sh/components/navigation/react-demos/drilldown-nav/  I do see that this is broken in the latest release of v4, so not sure when that happened. But we should fix for v5 if possible.

* There are two examples that still use the word "legacy." Can we change this? Perhaps change them to "Tertiary (deprecated)" and "Light nav (deprecated)." Or should we just remove these examples if we don't want demos that use deprecated components. @tlabaj @nicolethoen @mcoker thoughts?

**Comment by tlabaj** [comment]
> User menu looks good now. Just seeing a couple of additional issues.
> 
> * The drilldown nav example seems to be broken: https://patternfly-react-pr-9223.surge.sh/components/navigation/react-demos/drilldown-nav/  I do see that this is broken in the latest release of v4, so not sure when that happened. But we should fix for v5 if possible.
> * There are two examples that still use the word "legacy." Can we change this? Perhaps change them to "Tertiary (deprecated)" and "Light nav (deprecated)." Or should we just remove these examples if we don't want demos that use deprecated components. @tlabaj @nicolethoen @mcoker thoughts?

@mcarrano there is an issue open for drilldown nav here https://github.com/patternfly/patternfly-react/issues/9183

I updated the demo titles

**Review by mcarrano — APPROVED** [review]
Thanks @tlabaj . I am good to merge this.

**Review by kmcfaul — APPROVED** [review]
lgtm. 

I did notice that the nav drilldown demo looks like it may be broken, but this PR didn't touch that so we may want a follow up issue.

**Pull request merged**